exchange.go

```diff
@@ -352,3 +333,31 @@ func (e *Exchange) DataTransfer() datatransfer.Manager {
 func (e *Exchange) Supply() supply.Manager {
 	return e.supply
 }
+
+// StoragePeerInfo resolves a Filecoin address to find the peer info and add to our address book
+func (e *Exchange) StoragePeerInfo(ctx context.Context, addr address.Address) (*peer.AddrInfo, error) {
```

Comment:
Maybe just `StorePeerInfo`?

Reply:
Storage actually means the peer is a Storage provider, it's a special type of peer.